widget/entry.go

```diff
@@ -170,35 +170,41 @@ func (e *Entry) Enable() { // TODO remove this override after ReadOnly is remove
 
 // ExtendBaseWidget is used by an extending widget to make use of BaseWidget functionality.
 func (e *Entry) ExtendBaseWidget(wid fyne.Widget) {
```

Comment:
Note: this function has slightly different behaviour now, before it would only call registerShortcut() if e.BaseWidget.impl was nil. Now it calls registerShortcut() every time. Tho I don't think that is an issue based on how ShortcutHandler is implemented - it will just replace the old shortcuts with the identical new ones.

Reply:
That is a good point. This takes a bit of thinking but I think overall I should change how baseimpl is locked. In this instance the method may be called a lot so the write lock is probably a bad idea if not truly required.